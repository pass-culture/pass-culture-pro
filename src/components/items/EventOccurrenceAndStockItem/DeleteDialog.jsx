import React, { Component } from 'react'

class DeleteDialog extends Component {
  render() {
    const {
      isStockOnly,
      onCancelDeleteClick,
      onConfirmDeleteClick,
    } = this.props

    return (
      <tr>
        <td colSpan="6" className="is-size-7">
          En confirmant l'annulation de{' '}
          {isStockOnly ? 'ce stock' : 'cette date'}, vous supprimerez aussi
          toutes les réservations associées. {!isStockOnly && <br />}
          Êtes-vous sûr de vouloir continuer&nbsp;?
        </td>

        <td>
          <button className="button is-primary" onClick={onConfirmDeleteClick}>
            Oui
          </button>
        </td>
        <td>
          <button className="button is-primary" onClick={onCancelDeleteClick}>
            Non
          </button>
        </td>
      </tr>
    )
  }
}

export default DeleteDialog
