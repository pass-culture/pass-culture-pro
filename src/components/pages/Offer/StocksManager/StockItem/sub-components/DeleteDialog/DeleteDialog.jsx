import React, { Component } from 'react'

class DeleteDialog extends Component {
  render() {
    const { isEvent, onCancelDeleteClick, onConfirmDeleteClick } = this.props

    return (
      <tr>
        <td colSpan="6" className="is-size-7">
          En confirmant l'annulation de {isEvent ? 'cette date' : 'ce stock'},
          vous supprimerez aussi toutes les réservations associées.{' '}
          {isEvent && <br />}
          Êtes-vous sûr de vouloir continuer&nbsp;?
        </td>

        <td>
          <button className="button is-primary" onClick={onConfirmDeleteClick}>
            Oui
          </button>
        </td>
        <td>
          <button className="button is-primary" onClick={onCancelDeleteClick}>
            Non
          </button>
        </td>
      </tr>
    )
  }
}

export default DeleteDialog
