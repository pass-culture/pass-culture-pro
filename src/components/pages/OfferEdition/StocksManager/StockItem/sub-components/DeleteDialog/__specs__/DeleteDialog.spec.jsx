import { shallow } from 'enzyme'
import React from 'react'

import DeleteDialog from '../DeleteDialog'

describe('components | OfferEdition | DeleteDialog', () => {
  describe('snapshot', () => {
    it('should match the snapshot when isEvent = true', () => {
      // given
      const initialProps = {
        isEvent: true,
      }

      // when
      const wrapper = shallow(<DeleteDialog {...initialProps} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot when isEvent = false', () => {
      // given
      const initialProps = {
        isEvent: false,
      }

      // when
      const wrapper = shallow(<DeleteDialog {...initialProps} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })
  })
})
