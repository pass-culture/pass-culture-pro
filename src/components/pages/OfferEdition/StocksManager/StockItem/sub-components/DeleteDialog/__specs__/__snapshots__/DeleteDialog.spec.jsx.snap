// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | OfferEdition | DeleteDialog snapshot should match the snapshot when isEvent = false 1`] = `ShallowWrapper {}`;

exports[`components | OfferEdition | DeleteDialog snapshot should match the snapshot when isEvent = true 1`] = `ShallowWrapper {}`;
